src/features/suggestProduct/SuggestProduct.tsx

```diff
@@ -0,0 +1,126 @@
+import {useFormik} from 'formik';
+import * as Yup from 'yup';
+import {useQuery} from "@tanstack/react-query";
+import {getAllCategories} from "../../services/api.ts";
+import {useNavigate} from "react-router-dom";
+
+interface ProductSuggestion {
```

Comment:
Might be worth separating the types outside of this so you can use the type in state mgt etc.